File: Android/app/src/main/java/com/pokeapi/lpiem/pokeapiandroid/Provider/Pokemon/InterfaceCallBackController.java
package com.pokeapi.lpiem.pokeapiandroid.Provider.Pokemon;

import com.pokeapi.lpiem.pokeapiandroid.Model.Pokemon.Retrofit.PokemonRetrofit;

public interface InterfaceCallBackController<T>  {
    public void onWorkDone();
    public void showPokemon(PokemonRetrofit p);
}

File: Android/app/src/main/java/com/pokeapi/lpiem/pokeapiandroid/Provider/Pokemon/InterfaceCallBackController.kt
package com.pokeapi.lpiem.pokeapiandroid.Provider.Pokemon

import com.pokeapi.lpiem.pokeapiandroid.Model.Pokemon.Retrofit.PokemonRetrofit
import com.pokeapi.lpiem.pokeapiandroid.Model.Pokemon.Retrofit.Species

interface InterfaceCallBackController<T> {
    fun addPokemonSpecies(i: Int, s:Species)
    fun addPokemonToList(p: PokemonRetrofit)
    fun addPokedexEntry(i: Int, s:String)
}
